app/lib/methods/getFilePathAudio.ts

```diff
@@ -0,0 +1,20 @@
+import { getUserSelector } from '../../selectors/login';
+import { store } from '../store/auxStore';
+import { getFilePath } from './handleMediaDownload';
+import { IAttachment } from '../../definitions/IAttachment';
+
+export const getFilePathAudio = (file: IAttachment): string => {
+	const baseUrl = store.getState().server.server;
+	const cdnPrefix = store.getState().settings.CDN_PREFIX;
+	const user = getUserSelector(store.getState());
+
+	let url = file.audio_url;
+	if (url && !url.startsWith('http')) {
+		url = `${cdnPrefix || baseUrl}${url}`;
+	}
+	return getFilePath({
+		urlToCache: `${url}?rc_uid=${user.id}&rc_token=${user.token}`,
+		type: 'audio',
+		mimeType: file.audio_type
+	}) as string;
+};
```

Comment:
Hey @jsathu07, I believe this is the final part that you can enhance, and I had this idea a few hours ago. Could you create a custom hook with this function? By having a custom hook, we can replace how we get the file path in message/Audio.tsx, allowing us to reuse the code and prevent duplication between this file and the component.

Reply:
Sure sir

Reply:
@reinaldonetof Do I need to call this function inside a custom hook and straightaway pass the 'path' to downloadMediaFile?

Reply:
https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/6122c71d9760964c066779371bd6451648971d5a/app/containers/message/Audio.tsx#L37-L43
Here we are following the same approach as you. If you create a custom hook for it, that hook could replace these lines. In my humble opinion, you can proceed with this suggestion (please create a new folder within the message's folder to house the custom hook, placing it near the location where it will be used).


```
const useGetFilePathAudio = ({ audioUrl, audioType }: { audioUrl: string; audioType: string }) => {
	const [filePath, setFilePath] = useState('');

	const { cdnPrefix, baseUrl, user } = useAppSelector(state => ({
		cdnPrefix: state.settings.CDN_PREFIX as string,
		baseUrl: state.server.server,
		user: getUserSelector(state)
	}));

	useEffect(() => {
		const url = getFilePathAudio({...});
		setFilePath(url);
	}, [...]);

	return filePath;
};

export const getFilePathAudio = ({
	audioType,
	audioUrl,
	baseUrl,
	cdnPrefix,
	id,
	token
}: {
	audioUrl: string;
	audioType: string;
	cdnPrefix: string;
	baseUrl: string;
	id: string;
	token: string;
}) => {
	if (audioUrl && !audioUrl.startsWith('http')) {
		audioUrl = `${cdnPrefix || baseUrl}${audioUrl}`;
	}
	return getFilePath({
		urlToCache: `${audioUrl}?rc_uid=${id}&rc_token=${token}`,
		type: 'audio',
		mimeType: audioType
	}) as string;
};
```

Additionally, to use it inside the AudioPlayer class, you can follow the same approach you're currently using.

Please note that this block of code hasn't been tested and may be missing some elements. Could you test it and let us know if this approach is feasible? Thanks a lot @jsathu07